### PR TITLE
PAYARA-3805 Cannot edit, add, or delete JVM options from Admin Console

### PR DESCRIPTION
**Pull request by AlanRoth**
The problem was caused by '-XX:NewRatio=2' as previously it was assumed -X command-line options didn't require a value like in -D command-line options.

### CODE REVIEW

File: nucleus/admin/rest/rest-service/src/main/java/org/glassfish/admin/rest/resources/CollectionLeafResource.java [hunk 1]
```diff
@@ -352,10 +352,10 @@ private String getErrorMessage(Map<String, String> data, ActionReport ar) {
                 results.put(key, entry.getValue());
             } else {
                 options.append(sep).append(removeVersioning ? new JvmOption(key).option : key);
-
+
```
Comment (cubastanley) [inline nucleus/admin/rest/rest-service/src/main/java/org/glassfish/admin/rest/resources/CollectionLeafResource.java]: tabz

**Comment by Pandrex247** [comment]
Jenkins test please

File: nucleus/admin/rest/rest-service/src/main/java/org/glassfish/admin/rest/resources/CollectionLeafResource.java (hunk 1 above)
Comment (AlanRoth) [reply nucleus/admin/rest/rest-service/src/main/java/org/glassfish/admin/rest/resources/CollectionLeafResource.java]: Spaces ;)

File: nucleus/admin/rest/rest-service/src/test/java/fish/payara/admin/rest/resources/CollectionLeafResourceTest.java [hunk 2]
```diff
@@ -122,4 +122,29 @@ public void when_property_is_client_null_value_expect_add_to_map(){
         data8.put("-client", null);
         assertEquals(wrapper.getProcessedData(data8).get("id"), "-client");
     } 
+    
+    @Test
+    public void when_property_is_Xproperty_with_value_expect_add_to_map(){
+        Wrapper wrapper = new Wrapper();
+        Map<String, String> data9 = new HashMap<>();
+        data9.put("-XX:NewRatio=", "2");
+        assertEquals(wrapper.getProcessedData(data9).get("id"), "-XX:NewRatio=2");
```
Comment (jbee) [inline nucleus/admin/rest/rest-service/src/test/java/fish/payara/admin/rest/resources/CollectionLeafResourceTest.java]: I think expected and actual are swapped here (and for the other tests).

**Comment by Pandrex247** [comment]
@jbee It's using TestNg assert equals rather than Junit assert equals - they take _expected_ and _actual_ in reverse of each other (which is just lovely and will never lead to confusion)

File: nucleus/admin/rest/rest-service/src/test/java/fish/payara/admin/rest/resources/CollectionLeafResourceTest.java (hunk 2 above)
Comment (AlanRoth) [reply nucleus/admin/rest/rest-service/src/test/java/fish/payara/admin/rest/resources/CollectionLeafResourceTest.java]: TestNG assert equals just likes to be different - caught me out the first time. 😄

**Pull request merged**